cellrank/tl/estimators/mixins/decomposition/_eigen.py

```diff
@@ -0,0 +1,378 @@
+from typing import Any, Dict, Tuple, Union, Mapping, Optional
+from typing_extensions import Literal
+
+from types import MappingProxyType
+from pathlib import Path
+
+from anndata import AnnData
+from cellrank import logging as logg
+from cellrank._key import Key
+from cellrank.ul._docs import d
+from cellrank.tl._utils import save_fig, _eigengap
+from cellrank.tl.estimators._utils import SafeGetter
+from cellrank.tl.estimators.mixins._utils import BaseProtocol, logger, shadow
+
+import numpy as np
+from scipy.sparse import issparse, spmatrix
+from scipy.sparse.linalg import eigs
+
+import matplotlib.pyplot as plt
+from matplotlib.ticker import MultipleLocator, FormatStrFormatter
+
+EPS = np.finfo(np.float64).eps
+
+
+class EigenProtocol(BaseProtocol):  # noqa: D101
+    @property
+    def transition_matrix(self) -> Union[np.ndarray, spmatrix]:  # noqa: D102
+        ...
+
+    def _write_eigendecomposition(
+        self, decomp: Dict[str, Any], params: Mapping[str, Any] = MappingProxyType({})
+    ) -> str:
+        ...
+
+
+class EigenMixin:
+    """Mixin that computes Schur decomposition."""
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+
+        self._eigendecomposition = None
+
+    @property
+    @d.get_full_description(base="eigen")
+    def eigendecomposition(self) -> Optional[Dict[str, Any]]:
+        """
+        Eigendecomposition of :attr:`transition matrix`.
```

Comment:
```suggestion
        """Eigendecomposition of :attr:`transition matrix`.
```

Reply:
As I've said earlier, I won't do these sort of changes, because:
1. I like the it currently is
2. for consistency, we'd have to change it everywhere
3. doesn't interfere with doc building

Therefore am going to close/resolve all such comments.